### PR TITLE
[PHYSICS - DeleteRope]: Added Native Documentation 

### PR DESCRIPTION
**Pull request by JayPaulinCodes**
## Summary
Added documentation to the DeleteRope native with inspiration from the DeleteEntity native due to their similar function.

## Validating Example(s)
- Video showcasing the different types of ropes being spawned and deleted: https://youtu.be/hATpDNbu3bc
- Video showcasing the deleting portion of the documented Lua example: https://youtu.be/sCmscpspFIU
- Code used for the validating examples:
```lua
local rope1 = nil
local rope2 = nil

local function spawnRopeOfType(ropeType)
    -- Get the handle for the player's ped
    local pedHandle = PlayerPedId()

    -- Ensure that the rope textures are loaded
    RopeLoadTextures()
    while not RopeAreTexturesLoaded() do
        Citizen.Wait(0)
    end

    -- Get the coordinates for where the rope will be
    local ropePos = GetOffsetFromEntityInWorldCoords(pedHandle, 0.0, 2.0, 0.5)

    -- Create the rope
    local newRopeHandle = AddRope(ropePos.x, ropePos.y, ropePos.z, 0.0, 0.0, 0.0, 10.0, ropeType, 10.0, 0.0, 1.0, false, false, false, 1.0, false, 0)

    return newRopeHandle
end

RegisterCommand("deleterope_dev_spawn_1", function(soruce, args)
    rope1 = spawnRopeOfType(0)
    print("deleterope_dev_spawn_1 - post", rope1)
end)

RegisterCommand("deleterope_dev_delete_1", function(soruce, args)
    print("deleterope_dev_delete_1 - pre", rope1)
    local delResult = DeleteRope(rope1)
    print("deleterope_dev_delete_1 - post", rope1, delResult)
    rope1 = nil
end)

RegisterCommand("deleterope_dev_spawn_2", function(soruce, args)
    rope2 = spawnRopeOfType(3)
    print("deleterope_dev_spawn_2 - post", rope2)
end)

RegisterCommand("deleterope_dev_delete_2", function(soruce, args)
    print("deleterope_dev_delete_2 - pre", rope2)
    local delResult = DeleteRope(rope2)
    print("deleterope_dev_delete_2 - post", rope2, delResult)
    rope2 = nil
end)

RegisterCommand("deleterope_example", function()
    
    Citizen.CreateThread(function()
    
        -- Get the handle for the player's ped
        local pedHandle = PlayerPedId()

        -- Ensure that the rope textures are loaded
        RopeLoadTextures()
        while not RopeAreTexturesLoaded() do
            Citizen.Wait(0)
        end

        -- Get the coordinates for where the rope will be
        local ropePos = GetOffsetFromEntityInWorldCoords(pedHandle, 0.0, 2.0, 0.5)

        -- Create the rope
        local newRopeHandle = AddRope(ropePos.x, ropePos.y, ropePos.z, 0.0, 0.0, 0.0, 10.0, ropeType, 10.0, 0.0, 1.0, false, false, false, 1.0, false, 0)

        -- Delay so we can see the rope for a bit
        Citizen.Wait(1000)

        -- Check if the rope exists.
        if not DoesRopeExist(newRopeHandle) then 
            -- If the rope does not exist, end the execution of the code here.
            return 
        end
    
        -- If the rope does exist, delete the rope.
        DeleteRope(newRopeHandle)

    end)

end)
```

## Contribution Guidelines Acknowledgement
Before you submit this PR, please make sure:
- [x] You have read the contribution guidelines
- [x] You include an example that validates your change
- [x] Your English is grammatically correct

### CODE REVIEW

File: PHYSICS/DeleteRope.md [hunk 1]
```diff
@@ -5,10 +5,45 @@ ns: PHYSICS
...
+-- If the rope does exist, delete the rope.
+DeleteRope(rope)
+```
+
+```cs
+using CitizenFX.Core;
+using static CitizenFX.Core.Native.API;
+
+// Create a rope and store the reference
+Rope rope = World.AddRope(RopeType.ThinMetalWire, new Vector3(0f, 0f, 0f), new Vector3(0f, 0f, 0f), 0f, 0f, false);
```
Comment (AvarianKnight) [inline PHYSICS/DeleteRope.md:37]: We shouldn't use C# wrappers in the docs, or if we do we should have another example below it, see https://github.com/citizenfx/natives/blob/master/ENTITY/DoesEntityExist.md for an example on how this should be done.

I would just recommend using the native version.

**Review by AvarianKnight — CHANGES_REQUESTED** [review]

File: PHYSICS/DeleteRope.md (hunk 1 above)
Comment (JayPaulinCodes) [reply PHYSICS/DeleteRope.md:37]: Sounds good, thanks for the input!

**Review by AvarianKnight — CHANGES_REQUESTED** [review]
Some things to get the examples to work out of the box

File: PHYSICS/DeleteRope.md [hunk 2]
```diff
@@ -5,10 +5,44 @@ ns: PHYSICS
...
+
+// Check if the rope exists.
+if (!DoesRopeExist(ref ropehandle))
+{
+    // If the rope does not exist, end the execution of the code here.
+    return;
+}
+
+// If the rope does exist, delete the rope.
+DeleteRope(ref ropehandle);
```
[the remark this answers is not in the document]
Comment (JayPaulinCodes) [reply PHYSICS/DeleteRope.md:69]: This seems like a lot to put into an example, especially when compared to some of the other documentation examples I check prior to making this, is the standard to use a full script example, or a snippet of the code showcasing the use of the native in question? Just seems like there is much more here then elsewhere on C# examples.

[the remark this answers is not in the document]
Comment (AvarianKnight) [reply PHYSICS/DeleteRope.md:69]: In this case it was hard not to.

You'll notice most other C# examples don't ever interact with Tasks (besides [SET_PED_EMISSIVE_INTENSITY](https://docs.fivem.net/natives/?_0x4E90D746056E273D)) or have to wait for models to load.

It is possible to use SET_PED_EMISSIVE_INTENSITYs style, I'll discuss this some with some of the C# users in the community.

[the remark this answers is not in the document]
Comment (JayPaulinCodes) [reply PHYSICS/DeleteRope.md:69]: I see, once you discuss it with some other feel free to apply whichever way the general consensus is in favour of, or even a modified one. I see where your coming from however I do still feel the suggested example is a lot, I could agree to something more like what you mentioned with everything in 1 function though as to keep things cleaner, while in good programing breaking the cleanup and load jobs into their own function is the right thing to to, I feel in this case it adds too much clutter which isn't directly related to the example, and for the sake of the example, could probably be slotted into the 1 function.

File: PHYSICS/DeleteRope.md [hunk 3]
```diff
@@ -8,7 +8,120 @@ ns: PHYSICS
...
+	end
+	-- Let the rope exist for 3 seconds
+	Wait(3000)
+	-- Delete the rope!
+	DeleteRope(rope)
+	cleanup_rope_textures()
+end)
+```
+
+```cs
```
Comment (AvarianKnight) [inline PHYSICS/DeleteRope.md:54]:
```suggestion
```cs
using static CitizenFX.Core.Native.API;

```

**Pull request merged**